### PR TITLE
Configure Conscrypt Trust Manager

### PR DESCRIPTION
**Pull request by yschimke**
Cleanup version logic, introduce a minimum of Conscrypt 1.4.0 for version support

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -241,7 +241,7 @@ workflows:
       - testconscrypt:
           filters:
             branches:
-              only: master
+              ignore: master
```
Comment (yschimke) [inline .circleci/config.yml]: For PR testing

**Comment by yschimke** [comment]
cc @flooey

**Comment by yschimke** [comment]
Need to dig into the cause, so many bloody combinations :(

```
okurl --debug https://google.com/robots.txt
...
11:52:46.272	OkHttp Platform: ConscryptPlatform
11:52:46.273	Protocol: h2
11:52:46.273	TLS Version: TLS_1_3
11:52:46.273	Cipher: TLS_AES_128_GCM_SHA256
11:52:46.274	Peer Principal: CN=www.google.com, O=Google LLC, L=Mountain View, ST=California, C=US
11:52:46.274	Local Principal: none
11:52:46.274	JVM: 11.0.2+9
```

**Comment by yschimke** [comment]
Passing tests

https://circleci.com/gh/square/okhttp/760?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by swankjesse** [comment]
Why not require Conscrypt 2.1?

**Comment by swankjesse** [comment]
That avoids a lot of conditional code.

**Comment by yschimke** [comment]
Fair question, mainly because it was an optional dependency and it seemed like users were often using a combination of libraries that used Conscrypt.  But yes, happy to change, especially with OkHttp 4.0

**Comment by swankjesse** [comment]
Nice.

**Pull request merged**